specs/kyber/src/ring.rs

```diff
@@ -0,0 +1,122 @@
+use crate::parameters;
+use crate::field::FieldElement;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub(crate) struct RingElement {
+    pub coefficients : [FieldElement; parameters::COEFFICIENTS_IN_RING_ELEMENT]
+}
+
+impl RingElement {
```

Comment:
Some of this should probably not go to the library but needs some more comments in this case.
The `ntt_representation` is very much part of the spec.
But the `add`, `mul`, `multiply_matrix_by_vector` could probably go into the library as they are not spec specific.